Make worker cancellation a worker-native feature (#38752)
sourcegraph/sourcegraph: doc/dev/background-information/workers.md

This behavior can be controlled by setting the `StalledMaxAge` and `MaxNumResets` options on the database-backed store instance, which control the maximum grace period setting a record to _processing_ and locking it and number of times a record can be reset (to avoid poison messages from indefinitely crashing workers), respectively. Once a record hits the maximum number of resets, the resetter will move it from state _processing_ to _failed_ with a canned failure message.

### Cancelation

Cancelation of jobs in the database-backend store can be achieved in two ways:
1. By removing the job record from the database. The worker will eventually notice that the record doesn't exist anymore and will stop execution.

> 1. By setting `cancel` to `TRUE` on the record. Paired with the worker store setting `CancelInterval` will check for records to be canceled. These will ultimately end up in state `'canceled'`. This can be used to keep the record while still being able to cancel workloads.

Reviewer: ```suggestion
1. By setting `cancel` to `TRUE` on the record. If `CancelInterval` is set on the worker store, it will check for records to be canceled. These will ultimately end up in state `'canceled'`. This can be used to keep the record while still being able to cancel workloads.
```
